chips/cc26x0/src/uart.rs

```diff
@@ -74,12 +117,40 @@ impl UART {
     pub const fn new() -> UART {
         UART {
             regs: UART_BASE as *mut Registers,
+
             client: Cell::new(None),
-            tx_pin: Cell::new(None),
+            
+            rx_buffer: kernel::common::take_cell::TakeCell::empty(),
+            tx_buffer: kernel::common::take_cell::TakeCell::empty(),
+            
+            rx_remaining_bytes: Cell::new(0),
+            tx_remaining_bytes: Cell::new(0),
+
             rx_pin: Cell::new(None),
+            tx_pin: Cell::new(None),
+
+            offset: Cell::new(0),
         }
     }
 
+    pub fn configure_dma(&self) {
+        unsafe{
```

Comment:
```
unsafe {
       // Do unsafe stuff
 }
```
To make the unsafe block more explicit use indentation. No need for two separate unsafe by the way, imo it would be better to make the entire function `configure_dma` as unsafe instead